.teamcity/src/main/kotlin/promotion/PublishGradleDistribution.kt

```diff
@@ -34,26 +34,24 @@ abstract class PublishGradleDistribution(
 
     init {
         artifactRules = """
-        incoming-build-receipt/build-receipt.properties => incoming-build-receipt
         **/build/git-checkout/subprojects/base-services/build/generated-resources/build-receipt/org/gradle/build-receipt.properties
         **/build/distributions/*.zip => promote-build-distributions
         **/build/website-checkout/data/releases.xml
         **/build/git-checkout/build/reports/integTest/** => distribution-tests
         **/smoke-tests/build/reports/tests/** => post-smoke-tests
     """.trimIndent()
 
+        val absoluteId = RelativeId("Check_Stage_${this@PublishGradleDistribution.triggerName}_Trigger")
         steps {
             gradleWrapper {
                 name = "Promote"
                 tasks = task
-                gradleParams = """-PuseBuildReceipt $extraParameters "-PgitUserName=$gitUserName" "-PgitUserEmail=$gitUserEmail"  """
+                gradleParams = """-PcommitId=%dep.Gradle_${promotedBranch.capitalize()}_Check_Stage_${this@PublishGradleDistribution.triggerName}_Trigger.build.vcs.number% $extraParameters "-PgitUserName=$gitUserName" "-PgitUserEmail=$gitUserEmail"  """
             }
         }
+
         dependencies {
-            artifacts(RelativeId("Check_Stage_${this@PublishGradleDistribution.triggerName}_Trigger")) {
-                buildRule = lastSuccessful(promotedBranch)
-                cleanDestination = true
-                artifactRules = "build-receipt.properties => incoming-build-receipt/"
+            snapshot(RelativeId("Check_Stage_${this@PublishGradleDistribution.triggerName}_Trigger")) {
             }
```

Comment:
```suggestion
            snapshot(RelativeId("Check_Stage_${this@PublishGradleDistribution.triggerName}_Trigger"))
```

Let's remove the superfluous `{}`.

Reply:
Yeah.. that's my first feeling as well, but unfortunately `snapshot` method has no default parameter for that function...